### ISSUE TITLE
[conan-center] Hooks regarding `test_package` and `cpp_info`

### ISSUE THREAD

**Issue by jgsogo** [conan-io/hooks#374]
I'm proposing three hooks:

1. Every recipe should have a `test_package` folder (default name for Conan to run after `conan create`) (+info: https://github.com/conan-io/conan-center-index/pull/8389). 🤔  _Maybe it is already there_.
2. If `cpp_info` contains `.name` or `.filename`, then (one of) the generators `cmake_find_package[_multi]` are expected
3. If `cpp_info` contains `.set_property`, then (one of) the generators `CMakeDeps`, `MSBuildDeps`, `PkgConfigDeps` are expected (maybe depending on the property).

Not really sure about the last two entries, not sure if we want (so) many packages with several `test_xxxx` folders.

**Comment by uilianries** [comment]
> 1. Every recipe should have a test_package folder (default name for Conan to run after conan create) (+info: [poc] Test other generators: run several tests in test_*/conanfile.py conan-center-index#8389). thinking Maybe it is already there.

Yes, the Hook 24: TEST PACKAGE FOLDER: https://github.com/conan-io/hooks/blob/master/hooks/conan-center.py#L248

**Comment by uilianries** [comment]
> 3. If cpp_info contains .set_property, then (one of) the generators CMakeDeps, MSBuildDeps, PkgConfigDeps are expected (maybe depending on the property).

We can not adopt it now. We are "migrating" recipes on CCI, supporting both `cpp_info.names` and `set_property` at same recipe to be prepared for 2.0, however, new generators are not used yet. We could think about a warning message to consider refactoring the recipe to new generators support.
